das/avail/avail.go

```diff
@@ -0,0 +1,271 @@
+package avail
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"time"
+
+	gsrpc "github.com/centrifuge/go-substrate-rpc-client/v4"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/signature"
+	gsrpc_types "github.com/centrifuge/go-substrate-rpc-client/v4/types"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/types/codec"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/offchainlabs/nitro/arbutil"
+	"github.com/offchainlabs/nitro/das/avail/vectorx"
+	"github.com/offchainlabs/nitro/das/dastree"
+	"github.com/vedhavyas/go-subkey"
+)
+
+// AvailMessageHeaderFlag indicates that this data is a Blob Pointer
+// which will be used to retrieve data from Avail
+const AvailMessageHeaderFlag byte = 0x0a
+
+func IsAvailMessageHeaderByte(header byte) bool {
+	return (AvailMessageHeaderFlag & header) > 0
+}
+
+type AvailDA struct {
+	enable      bool
+	vectorx     vectorx.VectorX
+	timeout     time.Duration
+	appID       int
+	api         *gsrpc.SubstrateAPI
+	meta        *gsrpc_types.Metadata
+	genesisHash gsrpc_types.Hash
+	rv          *gsrpc_types.RuntimeVersion
+	keyringPair signature.KeyringPair
+	key         gsrpc_types.StorageKey
+}
+
+func NewAvailDA(cfg DAConfig, l1Client arbutil.L1Interface) (*AvailDA, error) {
+
```

Comment:
cleanup

Reply:
https://github.com/availproject/optimism-plasma-avail/blob/develop/op-plasma/avail/scripts/SubmitDataAndWatch.go

Reply:
`NewAvailDA` can't be divided into smaller functions, because or `NewAvailDA` function will not justify the structure of the `AvailDA` struct :- https://github.com/availproject/avail-nitro-adapter/pull/9/files#diff-135aa9566bddd6854e4e62dd4405662f9f36b2bf877463c9e7a11ee211ec5a40R31, As per good code design everything should be initialized separately here